### ISSUE TITLE
Jetpack Tiled Gallery: How should we handle (lack of) in-editor previews?

### ISSUE THREAD

**Issue by SiobhyB** [wordpress-mobile/gutenberg-mobile#4378]
The in-editor preview isn't going to be worked on until the second phase of the Tiled Gallery block. As such, users aren't going to see previews when they make changes to the block's styles or the rounded corner setting. This could be viewed as unintuitive, given that other blocks tend to display instant in-editor previews when settings are changed.

How should we navigate this until work on phase two of the block begins? For example, should we display a notice somewhere alongside the block? Any other ideas?

cc: @kyleaparker, @guarani, @illusaen

**Comment by guarani** [comment]
👋 @SiobhyB, there's a message shown in the designs to let users know how Mosaic and Columns are only shown in the page preview i.e. not in the in-editor preview, "The Mosaic style is only visible in the page preview": 

| <img width="300" alt="Screen Shot 2021-12-14 at 17 50 34" src="https://user-images.githubusercontent.com/1898325/146078520-b902124f-cbb0-4389-88f8-68f3fde917f3.png"> |
| - | 


> users aren't going to see previews when they make changes to the block's styles or the rounded corner setting

Our current proposal is to ship with in-editor previews of only the Square layout. That means we could show this message for all other layouts (Circle, Tiled Columns, Mosaic). 

---

About rounded corners, I don't recall whether or not we've decided to leave its in-editor preview out of the 1st release. Is there a discussion somewhere or an issue to track where we could decide how to handle it? Either way, I think we could keep that discussion in a separate issue and leave this one just for layout (block style) previews, would you agree?

**Comment by kyleaparker** [comment]
>How should we navigate this until work on phase two of the block begins? For example, should we display a notice somewhere alongside the block? Any other ideas?

I've rethought the approach of showing a message and think it would be best to hide these settings entirely. My suggestion would be to hide any setting that doesn't render an immediate preview. I think this is the least confusing approach to begin with, and we can then add the settings as they become available. 

The thinking is we won't confuse users who add this block from mobile with settings they can't use (previewing the post to see the change isn't intuitive), and I'm going to assume the number of people editing an existing tiled gallery block on mobile would be very small.

**Comment by SiobhyB** [comment]
@kyleaparker, @guarani, thanks both! I think I may have caused some confusion by creating a single issue for these two distinct settings/features: 

1. The different layouts i.e. square, circle, tiled, mosaic.
2. The rounded corner setting.

@kyleaparker, would your suggestion apply to both of the above or only the rounded corner setting?

If we hide the option of switching between the different layouts, would this mean shipping the block with only the square layout available? I believe there's currently some uncertainty around work on phase 2, so am also wondering how that'd play into that decision.

**Comment by mchowning** [comment]
> I believe there's currently some uncertainty around work on phase 2

To elaborate on that a bit from my perspective: _if_ we we feel that the block after Phase 1 is good enough to ship, I think it would be nice to hold off on Phase 2 in order to wait and see how much the JP Tiled Gallery Block is used, whether users want/need it to have more features (like in-editor preview) on it, etc. 

On the other hand, maybe we'll find that the Phase 1 block (i.e., without in-editor previews of the different layouts) isn't worth shipping, so we need to do Phase 2 now if we want to ship anything.

Just sharing my thoughts, this is a joint decision for all of us as far as I'm concerned.

**Comment by kyleaparker** [comment]
> @kyleaparker, would your suggestion apply to both of the above or only the rounded corner setting?

I'm suggesting both :) I think that way we can create a really great looking tiled gallery while not introducing any potentially confusing settings. I think this is especially important if we don't have a clear idea on when we would ship phase 2 as we wouldn't want the block to feel indefinitely unfinished.
